### ISSUE TITLE
Fix ant->maven antlr4 build regressions

### ISSUE THREAD

**Issue by arifogel** [batfish/batfish#195]
- [x] Change to an imported grammar file should trigger regeneration of grammar importing it
  Currently, only master parser/lexer grammar changes trigger a rebuild. Fixing this is **HIGHEST** priority

  - [x]  .tokens files should only be generated in target
  The .tokens files are now being generated in the src/main/antlr4/... directories. These should ideally not be produced in a src folder.

- [ ] Only recompile relevant java files when grammars are regenerated
  After any MASTER grammar file is changed, it appears that ALL java files are recompiled - rather than just the ones that have changed (or rely on ones that have changed). While this will certainly work, it seems heavy-handed. Please see behavior of old ant build.xml for a lighter-weight method.

- [ ] Parallel processing of grammars
Grammars are currently processed one at a time. When we used ant, all out-of-date lexers were generated in parallel, and then all out-of-date parsers were generated in parallel. While we could replicate this behavior, I'd like to do better:
  Each grammar should be its own task. Each parser task should depend on its corresponding lexer task. All such tasks should be executed in parallel to the extent feasible.

**Comment by dhalperi** [comment]
Re: tokens -- I believe that Maven only produces them in the `target` folder:

```
➜  batfish git:(master) find . -name \*.tokens
./projects/batfish/target/generated-sources/antlr4/AssertionLexer.tokens
./projects/batfish/target/generated-sources/antlr4/AssertionParser.tokens
./projects/batfish/target/generated-sources/antlr4/BatfishTopologyLexer.tokens
./projects/batfish/target/generated-sources/antlr4/BatfishTopologyParser.tokens
./projects/batfish/target/generated-sources/antlr4/CiscoLexer.tokens
./projects/batfish/target/generated-sources/antlr4/CiscoParser.tokens
./projects/batfish/target/generated-sources/antlr4/EosRoutingTableLexer.tokens
./projects/batfish/target/generated-sources/antlr4/EosRoutingTableParser.tokens
./projects/batfish/target/generated-sources/antlr4/FlatJuniperLexer.tokens
./projects/batfish/target/generated-sources/antlr4/FlatJuniperParser.tokens
./projects/batfish/target/generated-sources/antlr4/FlatVyosLexer.tokens
./projects/batfish/target/generated-sources/antlr4/FlatVyosParser.tokens
./projects/batfish/target/generated-sources/antlr4/GNS3TopologyLexer.tokens
./projects/batfish/target/generated-sources/antlr4/GNS3TopologyParser.tokens
./projects/batfish/target/generated-sources/antlr4/IptablesLexer.tokens
./projects/batfish/target/generated-sources/antlr4/IptablesParser.tokens
./projects/batfish/target/generated-sources/antlr4/JuniperLexer.tokens
./projects/batfish/target/generated-sources/antlr4/JuniperParser.tokens
./projects/batfish/target/generated-sources/antlr4/MrvLexer.tokens
./projects/batfish/target/generated-sources/antlr4/MrvParser.tokens
./projects/batfish/target/generated-sources/antlr4/NxosRoutingTableLexer.tokens
./projects/batfish/target/generated-sources/antlr4/NxosRoutingTableParser.tokens
./projects/batfish/target/generated-sources/antlr4/RoleLexer.tokens
./projects/batfish/target/generated-sources/antlr4/RoleParser.tokens
./projects/batfish/target/generated-sources/antlr4/VyosLexer.tokens
./projects/batfish/target/generated-sources/antlr4/VyosParser.tokens
```

Instead, my suspicion is that whatever IDE-based tools we use to edit ANTLR4 files may be producing the `.tokens` in the `src` tree next to the original `.g4` file.

**Comment by dhalperi** [comment]
I suspect we're at diminishing returns here – we've done the most important stuff and the remaining work is both hard and currently low ROI. So I'm going to close for now.
